editor/src/core/es-modules/package-manager/module-resolution.ts

```diff
@@ -221,15 +221,18 @@ function processPackageJson(
   return foldEither(
     (_) => resolveNotPresent,
     (packageJson) => {
-      const moduleName: string = packageJson.name ?? makePathFromParts(containerFolder)
+      const moduleName: string | null = packageJson.name ?? last(containerFolder) ?? null
       const mainEntry: string | null = packageJson.main ?? null
       const moduleEntry: string | null = packageJson.module ?? null
-      if (moduleEntry != null && mainEntry == null) {
-        return resolveSuccess(normalizePath([...containerFolder, ...getPartsFromPath(moduleName)]))
-      }
-      if (mainEntry != null) {
+      if (mainEntry != null && !mainEntry.endsWith('.cjs')) {
         return resolveSuccess(normalizePath([...containerFolder, ...getPartsFromPath(mainEntry)]))
       }
+      if (moduleEntry != null && !moduleEntry.endsWith('.cjs')) {
+        return resolveSuccess(normalizePath([...containerFolder, ...getPartsFromPath(moduleEntry)]))
+      }
+      if (moduleName != null) {
```

Comment:
this is surprising here, are we saying we resolve modules that only have a name field? is that according to the spec?

Reply:
Module name here is falling back to the directory, but this is actually missing a follow up commit that makes that clearer - will push that when I'm back at my keyboard

Reply:
Fixed in the latest commit